pkg/vcs/git.go

```diff
@@ -102,10 +101,9 @@ func (git *git) CheckoutBranch(repo, branch string) (*Commit, error) {
 	if err := git.repair(); err != nil {
 		return nil, err
 	}
-	repoHash := hash.String([]byte(repo))
 	// Ignore error as we can double add the same remote and that will fail.
-	git.git("remote", "add", repoHash, repo)
-	_, err := git.git("fetch", "--force", repoHash, branch)
+	git.git("remote", "add", "origin", repo)
```

Comment:
In some cases we add multiple remotes, they need to have unique names.
Is there another way to this the issue?

Reply:
That makes sense, thanks! There should be a way to set a remote for the submodules as well; I'll look into it.

Reply:
I'm unfortunately struggling to add submodules without 'origin'.
From the documentation, 
`The default remote is the remote of the remote-tracking branch of the current branch. If no such remote-tracking branch exists or the HEAD is detached, "origin" is assumed to be the default remote. If the superproject doesn’t have a default remote configured the superproject is its own authoritative upstream and the current working directory is used instead.`
Because this function initializes a detached head state, it reverts to the default remote.
The options I've found to fix this are:
1. Adding a remote origin, as I did above. It gets initialized to the superproject, and as multiple remotes can be added the origin is only used in initializing submodules
2. Modifying the .gitmodules file. It currently has relative paths for the submodules, but if absolute paths are added they will be used instead of the default origin. However, they would have to be updated whenever the remote is switched, unless a different branch is used
3. Merging the branch (e.g. git pull) instead of working in a detached head state

As adding a remote origin still seems to be the most straightforward without changing functionality, I went with that option for now, but could definitely look into another option if that would be more helpful.

Reply:
Do we need just any random origin set? Or the one we are currently checking out?
I think this will not work to set the origin we are currently checking out. The repo has 20+ remotes on syzbot, and origin can be added only once (I assume repeated additions will fail). So the origin will be set to some random non-deterministic repo that happened to be used with CheckoutBranch first.

Reply:
Modifying .gitmodules looks hacky and complex.
I am not sure what effects will have changing git fetch to git pull. git is somewhat unpredictable in some corner cases (e.g. we have some force-pushed trees).
I think the simplest will be delete "origin" and then add again. But also add a comment as to why we are doing this.

Reply:
Thanks for looking through this; that makes sense. I'll delete the original "origin" as well, and add an explanitory comment.